content/artifacts/easy-highlight-artifacts/plugin.js

```diff
@@ -0,0 +1,426 @@
+
+  const planetTypeFilter = html`<div>
+    <div style=${{ marginBottom: "3px" }}>Planet Type</div>
+    <${CreateSelectFilter}
+      items=${PLANET_TYPES}
+      selectedValue=${selectedPlanetType}
+      onSelect=${setSelectedPlanetType}
+    />
+  </div>`;
+
+  const mustHoldArtifactsFilter = html`
+    <div>
+      <div style=${{ marginBottom: "3px" }}>Artifacts</div>
+      <input
+        type="checkbox"
+        title="planets must hold artifacts"
+        onChange=${setMustHoldArtifacts}
+        defaultChecked=${mustHoldArtifacts}
+      />
+    </div>
+  `;
+
+  const ownerFilter = html`
+    <div>
+      <div style=${{ marginBottom: "3px" }}>Owner</div>
+      <input
+        type="checkbox"
+        title="planet is claimed by you"
+        onChange=${() => setIsOwner(!isOwner)}
+        defaultChecked=${isOwner}
+      />
+    </div>
+  `;
+
+  const planetUnionFilters = html`
+    <div style=${{ ...flexRow, justifyContent: "space-between", marginTop: "10px" }}>
+      ${planetTypeFilter} ${mustHoldArtifactsFilter} ${ownerFilter}
+    </div>
+  `;
+
+  const artifactFilters = html`<div style=${{ ...flexRow, marginTop: "10px" }}>
+    <div>
+      <div style=${{ marginBottom: "3px" }}>Artifacts Type</div>
+      <${CreateSelectFilter}
+        items=${ARTIFACT_TYPES}
+        selectedValue=${selectedArtifactType}
+        onSelect=${(v) => {
+          setSelectedArtifactType(v);
+          console.log("level mm", selectedLevels);
+        }}
+      />
+    </div>
+    <div style=${{ marginLeft: "75px" }}>
+      <div style=${{ marginBottom: "3px" }}>Artifacts Rarity</div>
+      <${CreateSelectFilter}
+        items=${ARTIFACT_RARITIES}
+        selectedValue=${selectedArtifactRarity}
+        onSelect=${setSelectedArtifactRarity}
+      />
+    </div>
+  </div>`;
+
+  const [loading, setLoading] = useState(false);
+
+  const highlightPlanet = () => {
+    setLoading(true);
+    eligiblePlanets = getEligiblePlanets();
+    setLoading(false);
+  };
+
+  const submitButton = html`<${createButton}
+    loading=${loading}
+    onClick=${highlightPlanet}
+  />`;
+
+  return html`
+    ${planetLevelFilter} ${planetUnionFilters} ${artifactFilters}
+    <${createDivider} />
+    ${submitButton}
+  `;
+}
+
+//@ts-ignore
+class Plugin {
+  draw(ctx) {
+    // @ts-ignore
+    const viewport = ui.getViewport();
+
+    ctx.save();
+    ctx.fillStyle = "red";
+    ctx.strokeStyle = "red";
+    for (let planet of eligiblePlanets) {
+      if (!planet.location) continue;
+      let { x, y } = planet.location.coords;
+      ctx.beginPath();
+      ctx.arc(
+        viewport.worldToCanvasX(x),
+        viewport.worldToCanvasY(y),
+        viewport.worldToCanvasDist(
+          ui.getRadiusOfPlanetLevel(planet.planetLevel)
+        ),
+        0,
+        2 * Math.PI
+      );
+      ctx.fill();
+      ctx.stroke();
+      ctx.closePath();
+    }
+    ctx.restore();
+  }
+
+  async render(container) {
+    render(html`<${App} />`, container);
+  }
```

Comment:
This needs a destroy method that cleans up after preact

Reply:
You track the container as `this.container` and then you use `render(null, this.container)` in the destroy method